### PR TITLE
Y admin forms

### PR DESCRIPTION
**Pull request by SuccessMaheshwari**

### CODE REVIEW

File: app/controllers/animation_datas_controller.rb [hunk 1]
```diff
@@ -9,4 +9,31 @@ def show
...
+    @new_animation = AnimationData.new
+    @new_animation.animation_name = permit_params[:animation_name]
+    @new_animation.animation_price = permit_params[:animation_price]
+    @new_animation.picture = permit_params[:picture]
+    @new_animation.category_id = params[:sub_category_id]
+    file_data = permit_params[:animation_json]
+    json_data = file_data.read
+    @new_animation.animation_json = json_data.as_json
+    if @new_animation.save()
+      puts @new_animation.animation_json
```
Comment (PreetiDantkale) [inline app/controllers/animation_datas_controller.rb]: remove puts statment

File: app/controllers/animation_datas_controller.rb [hunk 2]
```diff
@@ -9,4 +9,31 @@ def show
...
+      redirect_to animations_path
+    else
+      redirect_to new_animation_path
+    end
+  end
+
+  private
+  def permit_params
```
Comment (PreetiDantkale) [inline app/controllers/animation_datas_controller.rb]: indentation

File: app/javascript/packs/demo_animation.js [hunk 3]
```diff
@@ -15,9 +16,10 @@ var anim = bodymovin.loadAnimation(animData);
 setTimeout(function () {
   var html = animData.container.getElementsByClassName("lottie")[0];
   html.style.position = "fixed";
-  html.style.zIndex = -999;
+  //html.style.zIndex = -999;
```
Comment (PreetiDantkale) [inline app/javascript/packs/demo_animation.js]: remove comments

File: app/controllers/Users/registrations_controller.rb [hunk 4]
```diff
@@ -60,7 +60,7 @@ def copy_file_contents(file_name,user)
     end
 
   def delete_user_file
-    File.delete("public/users/#{current_user.id}.js")
+    File.delete("public/users/#{current_user.id}.min.js")
```
Comment (PreetiDantkale) [inline app/controllers/Users/registrations_controller.rb]: File.delete("public/users/#{current_user.id}.min.js") while deleting , you are deleting min.js and 
while creation , file name is only .js

File: app/javascript/packs/demo_animation.js (hunk 3 above)
Comment (PreetiDantkale) [reply app/javascript/packs/demo_animation.js]: Indentation

File: app/controllers/animation_datas_controller.rb [hunk 5]
```diff
@@ -9,4 +9,31 @@ def show
...
   end
+
+  def new
+    @new_animation = AnimationData.new
+    authorize @new_animation
+  end
+
+  def create
+    @new_animation = AnimationData.new
+    @new_animation.animation_name = permit_params[:animation_name]
```
Comment (SagareGanesh) [inline app/controllers/animation_datas_controller.rb]: use proper new syntax

File: app/controllers/categories_controller.rb [hunk 6]
```diff
@@ -20,6 +20,18 @@ def show
...
+  def new_sub_category
+    @new_sub_category = Category.new
+    authorize @new_sub_category
+    @primary_category_id = params[:id]
+  end
+
+  def create_sub_category
+    @new_sub_category = Category.new(permit_params)
+    @new_sub_category.primarycategory_id = params[:id]
+    @new_sub_category.save()
```
Comment (SagareGanesh) [inline app/controllers/categories_controller.rb]: no need () for save

File: app/policies/animation_data_policy.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class AnimationDataPolicy < ApplicationPolicy
+  attr_reader :user
+
+  def initialize(user, animation_datas)
+    @user = user
+    @animation_datas = animation_datas
+  end
+
+  def new?
+    Role.find_by(id: @user.role_id).role_name == "admin"
```
Comment (SagareGanesh) [inline app/policies/animation_data_policy.rb]: use user.role.name

File: app/policies/category_policy.rb [hunk 8]
```diff
@@ -10,4 +10,7 @@ def new?
     Role.find_by(id: @user.role_id).role_name == "admin"
   end
 
+  def new_sub_category?
+    Role.find_by(id: @user.role_id).role_name == "admin"
```
Comment (SagareGanesh) [inline app/policies/category_policy.rb]: same here

File: app/controllers/animation_datas_controller.rb (hunk 5 above)
Comment (SuccessMaheshwari) [reply app/controllers/animation_datas_controller.rb]: removed new

File: app/controllers/categories_controller.rb (hunk 6 above)
Comment (SuccessMaheshwari) [reply app/controllers/categories_controller.rb]: removed ()

File: app/policies/category_policy.rb (hunk 8 above)
Comment (SuccessMaheshwari) [reply app/policies/category_policy.rb]: Used 
role_name: @user.role.role_name

File: app/controllers/Users/registrations_controller.rb [hunk 9]
```diff
@@ -43,7 +43,7 @@ class Users::RegistrationsController < Devise::RegistrationsController
   # protected
   def create_user_file
     @user = current_user.id
-    @file_name = "public/users/#{@user}.js"
+    @file_name = "public/users/#{@user}.min.js"
```
Comment (PreetiDantkale) [inline app/controllers/Users/registrations_controller.rb]: no need of instance variables

File: app/javascript/packs/demo_animation.js [hunk 10]
```diff
@@ -1,4 +1,5 @@
 show = function(animation){
+  animation= JSON.parse(animation);
 	var animData = {
```
Comment (PreetiDantkale) [inline app/javascript/packs/demo_animation.js]: indentation

**Pull request merged**